life/life.py

```diff
@@ -4,15 +4,43 @@
 """
 
 import boards
+import sys
+
+def select_board():
+    boards_ = [boards.Glider, boards.LightWeightShip, boards.MiddleWeightShip, boards.HeavyWeightShip]
+    print("Boards: \n"
+          "1. Glider\n"
+          "2. Light Weight Ship\n"
+          "3. Middle Weight Ship\n"
+          "4. Heavy Weight Ship\n")
+
+    bd_index = int(input("Enter a corresponding number to choose a board: ")) - 1
+
+    if bd_index < 0 or bd_index > 3:
+        raise IndexError("Could not find a board with that index.")
+
+    def make_board(width, height):
+        print("Creating your board...")
+        board = boards_[bd_index](width, height)
+        print("Done!")
+        return board
+
+    return make_board
 
 def main():
-    board = boards.HeavyWeightShip(15, 15)
+    print("Welcome to Conway's Game of Life, select a board to start the game.")
+    make_board = select_board()
+    n = int(input("How many rows or columns should your board have? "))
+    board = make_board(n, n)
     gen = boards.next_gen(board)
+    nxt = 'next'
 
-    for _ in range(5):
-        print('\n----------------\n')
+    while nxt.lower() == 'next':
+        print('\n{}\n'.format('--' * board.height))
         print(next(gen))
+        nxt = input("< type next to see the next generation > OR < type anything you want to quit >")
+    sys.exit()
```

Comment:
I was not picturing this function like this in my mind. You did a great job! I wanted to bring all three generations for the current board setup then ask if they want to play with a mother one. However, I believe this is a much better design.

Just one thing, they cannot change the board. We should look for a way to let them change the board once they have decided to `exit` the game after typing anything other than `next`

Reply:
Thank you. Definitely, there should be a way for the user to restart the game and choose a new board.

Reply:
Also, should we make sure when they are giving the size they want the board to be, they will not give any number that cannot handle the cells they will be playing with!

For example, `HeavyWeightShip` cannot fit on a `5x5` board.